Lib/typing.py

```diff
@@ -1024,7 +1024,7 @@ def evaluate_forward_ref(
     owner=None,
     globals=None,
     locals=None,
-    type_params=_sentinel,
+    type_params=None,
```

Comment:
My concern is that it would be rather hard to miss. This function has multiple parameters, `__type_params__` is a new thing, which not many people are familiar with.

Proposed API makes it easy to miss.

And when missed, simple code like `-> T` and `evaluate_forward_ref('T', format=VALUE)` (instead of `evaluate_forward_ref('T', format=VALUE, type_params=func.__type_params__)`) can raise an error.

On the other hand, `type_params` is indeed optional in a sense, that users can't even provide `type_params` in all cases.

I have very mixed feelings about it :(

Reply:
In your example the user should rather pass `owner=`. Pointing them to use `type_params=` only makes the API more complicated.